src/main/java/org/isf/generaldata/DatabaseConfig.java

```diff
@@ -0,0 +1,10 @@
+package org.isf.generaldata;
+
+import org.springframework.context.annotation.Configuration;
+import org.springframework.context.annotation.PropertySource;
+
+@Configuration
+@PropertySource("classpath:database.properties")
```

Comment:
Tests can be fixed either adding the following property or adding database.properties to tests resources.
Of course the latter means that database.properties must be always present in the classpath, either from GUI or API.
Wdyt?

```suggestion
@PropertySource(value = "classpath:database.properties", ignoreResourceNotFound = true)
```

Reply:
Before, in the `applicationContext.xml` it was set to false:

```
	<bean
		class="org.springframework.beans.factory.config.PropertyPlaceholderConfigurer">
		<property name="location">
			<value>classpath:database.properties</value>
		</property>
		<property name="ignoreResourceNotFound" value="false" />
	</bean>
```

Reply:
I would lean towards the change in the `@PropertySource`

Reply:
I think that if we set not compulsory, it may lead to confusion and subtle behaviors... I've indeed discovered the problem because I was providing database.properties but was ignored...

Reply:
let's try with database.properties compulsory, so we can spot possible issues in other connected repositories

Reply:
I agree, in order to be consistent in every installation I would set the database.properties file as mandatory.

Reply:
Ok, then, I'm going to merge. Thanks!